### PR TITLE
Added support for "nested" mappings

### PR DESCRIPTION
**Pull request by petardz**
*Issue #, if available:*

*Description of changes:*
Fixed bug where PUT _mapping against query index would fail in case of "nested" properties in index mappings

*CheckList:*
[ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: alerting/src/main/kotlin/org/opensearch/alerting/util/DocLevelMonitorQueries.kt [hunk 1]
```diff
@@ -95,6 +100,22 @@ class DocLevelMonitorQueries(private val client: Client, private val clusterServ
...
     }
 
+    /**
+     * From given index mapping node, extracts fieldName -> fieldProperties pair
+     */
+    fun extractField(node: MutableMap<String, Any>, currentPath: String): Pair<String, MutableMap<String, Any>> {
+        if (node.containsKey(PROPERTIES)) {
+            return extractField(node.get(PROPERTIES) as MutableMap<String, Any>, currentPath)
+        } else if (node.containsKey(NESTED)) {
+            return extractField(node.get(NESTED) as MutableMap<String, Any>, currentPath)
```
Comment (lezzago) [inline alerting/src/main/kotlin/org/opensearch/alerting/util/DocLevelMonitorQueries.kt:110]: Is it possible for a node to contain `PROPERTIES` and `NESTED`?

Comment (petardz) [reply alerting/src/main/kotlin/org/opensearch/alerting/util/DocLevelMonitorQueries.kt:110]: I don't think so. They should be "lonely" interim nodes

File: alerting/src/test/kotlin/org/opensearch/alerting/MonitorDataSourcesIT.kt [hunk 2]
```diff
@@ -138,12 +139,24 @@ class MonitorDataSourcesIT : AlertingSingleNodeTestCase() {
...
+            PutMappingRequest(
+                index
+            ).source("test_alias.field_a", "type=alias,path=message")
+        ).get()
+        client().admin().indices().putMapping(
+            PutMappingRequest(
+                index
+            ).source("test_alias2", "type=alias,path=test_field")
+        ).get()
+
```
Comment (lezzago) [inline alerting/src/test/kotlin/org/opensearch/alerting/MonitorDataSourcesIT.kt:159]: Can we make a new test for nested situation?

File: alerting/src/main/kotlin/org/opensearch/alerting/util/DocLevelMonitorQueries.kt (hunk 1 above)
Comment (lezzago) [reply alerting/src/main/kotlin/org/opensearch/alerting/util/DocLevelMonitorQueries.kt:110]: Yea taking more of a look into that, seems to be right

**Review by lezzago — APPROVED** [review]

**Review by getsaurabh02 — APPROVED** [review]

**Review by praveensameneni — APPROVED** [review]

**Pull request merged**